.circleci/config.yml

```diff
@@ -8,22 +8,34 @@ jobs:
     parameters:
       node_version:
         type: string
+      package-lock:
+        type: string
     working_directory: ~/code-sdk
     docker:
       - image: cimg/node:<< parameters.node_version >>
     steps:
       - checkout
       - restore_cache:
           keys:
-            - v2-{{ .Branch }}-{{ .Revision }}-<< parameters.node_version >>
-      - run:
-          name: Install
-          command: npm install
+            - v2-{{ .Branch }}-{{ .Revision }}-<< parameters.node_version >>>-{{ checksum "package-lock.json" }}
+      - when:
+          condition: << parameters.package-lock >> == "locked-dependencies"
+          steps:
+            - run:
+                name: Install
+                command: npm install
+      - unless:
+          condition: << parameters.package-lock >>
+          steps:
+            - run:
+                name: Install without locked dependencies
+                command: npm install --no-package-lock
       - save_cache:
-          key: v2-{{ .Branch }}-{{ .Revision }}-<< parameters.node_version >>
+          key: v2-{{ .Branch }}-{{ .Revision }}-<< parameters.node_version >>-{{ checksum "package-lock.json" }}
```

Comment:
q: Is this correct when running with `--no-package-lock` or would this pollute the cache?

Reply:
This will pollute the cache. I'll substitute checksum with `package-lock` parameter for cache persistency.